pkg/kubelet/kubelet.go

```diff
@@ -1477,6 +1479,38 @@ func (kl *Kubelet) updateRuntimeUp() {
 	}
 }
 
+func (kl *Kubelet) reconcileCbr0Loop() {
+	if err := kl.reconcileCbr0(); err != nil {
+		glog.Errorf("Could not reconcile cbr0: %v", err)
+	}
+	for {
+		select {
+		case <-time.After(10 * time.Second):
+			if err := kl.reconcileCbr0(); err != nil {
+				glog.Errorf("Could not reconcile cbr0: %v", err)
+			}
+		}
+	}
+}
+
+func (kl *Kubelet) reconcileCbr0() error {
+	currentNode, err := kl.kubeClient.Nodes().Get(kl.hostname)
+	if currentNode.Spec.PodCIDR == "" {
+		glog.V(5).Info("PodCIDR not set. Will not configure cbr0.")
+		return nil
+	}
+	_, cidr, err := net.ParseCIDR(currentNode.Spec.PodCIDR)
+	if err != nil {
+		return err
+	}
+	// Set cbr0 interface address to first address in IPNet
+	cidr.IP.To4()[3] += 1
+	if err := ensureCbr0(cidr); err != nil {
```

Comment:
I don't see the "gross shelling out" or an implementation of ensureCbr0 in this PR. Did you miss part of the change?